### DISCUSSION TITLE
Can we hide the next tab based on the value of a field from previous tab?

### DISCUSSION THREAD

**Discussion by venkatesh1146** [manojadams/mui-forms#25] [Q&A] [answered]
I want to hide next tab based on the value of a field in the current tab. Is this supported in the current version @manojadams ?

**Comment by manojadams** [comment]
Currently we do have provision to enable tabs as follows:

```
"events": {
    "click": {
        "type": "emit-event",
        "value": "$enable-current-tab"
    }
}
```

But disable tab based on certain field value is currently not present. 

To understand your requirement better, can you give me a sample example with more details of your use case?
One sample example: https://codesandbox.io/p/sandbox/customer-feedback-form-mui-forms-63mlks

**Reply by venkatesh1146** [reply]
Hi @manojadams 

[Usecase code](https://codesandbox.io/p/devbox/sqctwx?file=%2Fsrc%2FCustomerFeedbackForm.jsx%3A18%2C23&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsbr373g00063b6ioa8bmqvz%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsbr373g00023b6idbyukvrd%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsbr373g00033b6iarr3470a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsbr373g00053b6impo56hot%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B58.642493476158315%252C41.357506523841685%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsbr373g00023b6idbyukvrd%2522%253A%257B%2522id%2522%253A%2522clsbr373g00023b6idbyukvrd%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsbsg1k600023b6k0hb4a5ot%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A18%252C%2522startColumn%2522%253A23%252C%2522endLineNumber%2522%253A18%252C%2522endColumn%2522%253A23%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FCustomerFeedbackForm.jsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsbsg1k600023b6k0hb4a5ot%2522%257D%252C%2522clsbr373g00053b6impo56hot%2522%253A%257B%2522id%2522%253A%2522clsbr373g00053b6impo56hot%2522%252C%2522activeTabId%2522%253A%2522clsbs6ke700793b6k9jt9mxyb%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clsbs6ke700793b6k9jt9mxyb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%255D%257D%252C%2522clsbr373g00033b6iarr3470a%2522%253A%257B%2522id%2522%253A%2522clsbr373g00033b6iarr3470a%2522%252C%2522activeTabId%2522%253A%2522clsbs6kbd00553b6k0hso7hwz%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clsbs6kbd00553b6k0hso7hwz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
Here is our requirement, I would like to hide the email field in second section when the countries is `IN`. 
Also I am not able to hide the second name based on the countries field.
What am i missing here ?

**Reply by manojadams** [reply]
Hi @venkatesh1146 ,

The key `dependencies` has to be inside `meta` object.
Can you check now?

For your reference: https://mui-forms.vercel.app/field-dependencies/exists#exists-dependency-example

**Reply by venkatesh1146** [reply]
Hi @manojadams 
Thank you for the quick update. 
Update my code working for same tab fields.

Any solution that we can use to hide the field in other tab/section ?

**Reply by manojadams** [reply, marked as answer]
To refer to fields in other tab's/section, you can use the key `section`. 

Check updated docs here:
https://mui-forms.vercel.app/field-dependencies/exists

For example:

```
dependencies: {
  exists: {
    section: "basic_details", // this refers to fields in other tab/group of the form
    ref: "countries",
    value: "IN"
  }
}
```
